<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>801</width>
    <height>649</height>
   </rect>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_2" stretch="0,1">
    <item>
     <widget class="QGroupBox" name="groupBoxDevices">
      <property name="title">
       <string>Discovered PicoScope</string>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QListWidget" name="deviceList"/>
       </item>
       <item>
        <layout class="QVBoxLayout" name="layoutDeviceButtons">
         <item>
          <widget class="QPushButton" name="pbConnect">
           <property name="text">
            <string>Connect</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="pbDisconnect">
           <property name="text">
            <string>Disconnect</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="pbRefresh">
           <property name="text">
            <string>Refresh</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="verticalSpacer">
           <property name="orientation">
            <enum>Qt::Orientation::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="groupBoxFrameSetup">
      <property name="title">
       <string>DMX</string>
      </property>
      <layout class="QHBoxLayout" name="frameSetupLayout" stretch="0,0">
       <item>
        <widget class="QGroupBox" name="groupBoxTiming">
         <property name="title">
          <string>Timing</string>
         </property>
         <layout class="QFormLayout" name="formLayout">
          <item row="0" column="0">
           <widget class="QLabel" name="frameTimeLabel">
            <property name="text">
             <string>Frame</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QSpinBox" name="frameTimeDisplay">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="buttonSymbols">
             <enum>QAbstractSpinBox::ButtonSymbols::NoButtons</enum>
            </property>
            <property name="suffix">
             <string notr="true"> μs</string>
            </property>
            <property name="maximum">
             <number>1000000</number>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="bitTimeLabel">
            <property name="text">
             <string>Bit</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QSpinBox" name="bitTimeEdit">
            <property name="suffix">
             <string notr="true"> μs</string>
            </property>
            <property name="minimum">
             <number>3</number>
            </property>
            <property name="maximum">
             <number>5</number>
            </property>
            <property name="value">
             <number>4</number>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="breakTimeLabel">
            <property name="text">
             <string>Break</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QSpinBox" name="breakTimeEdit">
            <property name="suffix">
             <string notr="true"> μs</string>
            </property>
            <property name="maximum">
             <number>1000000</number>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="mabTimeLabel">
            <property name="text">
             <string>MAB</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QSpinBox" name="mabTimeEdit">
            <property name="toolTip">
             <string>Mark After Break</string>
            </property>
            <property name="suffix">
             <string notr="true"> μs</string>
            </property>
            <property name="maximum">
             <number>1000000</number>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="mtbsTimeLabel">
            <property name="text">
             <string>MTBS</string>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="QSpinBox" name="mtbsTimeEdit">
            <property name="toolTip">
             <string>Mark Time Between Slots</string>
            </property>
            <property name="suffix">
             <string notr="true"> μs</string>
            </property>
            <property name="maximum">
             <number>1000000</number>
            </property>
           </widget>
          </item>
          <item row="5" column="0">
           <widget class="QLabel" name="mbbTimeLabel">
            <property name="text">
             <string>MBB</string>
            </property>
           </widget>
          </item>
          <item row="5" column="1">
           <widget class="QSpinBox" name="mbbTimeEdit">
            <property name="toolTip">
             <string>Mark Before Break</string>
            </property>
            <property name="suffix">
             <string notr="true"> μs</string>
            </property>
            <property name="maximum">
             <number>1000000</number>
            </property>
           </widget>
          </item>
          <item row="6" column="0" colspan="2">
           <widget class="QGroupBox" name="groupBoxPresets">
            <property name="title">
             <string>Presets</string>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout">
             <item>
              <widget class="QPushButton" name="pbPreset1986">
               <property name="text">
                <string>USITT DMX512:1986</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="pbPreset1990">
               <property name="text">
                <string>USITT DMX512:1990</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="pbPresetDMX512A">
               <property name="text">
                <string>ANSI E1.11:2004</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="pbPresetRDM">
               <property name="text">
                <string>ANSI E1.20:2006</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBoxContents">
         <property name="title">
          <string>Slots</string>
         </property>
         <property name="flat">
          <bool>false</bool>
         </property>
         <property name="checkable">
          <bool>false</bool>
         </property>
         <layout class="QGridLayout" name="gridLayout_2">
          <item row="0" column="1">
           <widget class="QGroupBox" name="groupBoxSlotPatterns">
            <layout class="QGridLayout" name="gridLayout">
             <property name="leftMargin">
              <number>0</number>
             </property>
             <property name="topMargin">
              <number>0</number>
             </property>
             <property name="rightMargin">
              <number>0</number>
             </property>
             <property name="bottomMargin">
              <number>0</number>
             </property>
             <item row="1" column="0">
              <widget class="QPushButton" name="pbSlotPatternZero">
               <property name="text">
                <string>All 0%</string>
               </property>
              </widget>
             </item>
             <item row="0" column="2">
              <widget class="QPushButton" name="pbSlotPatternIncreasing">
               <property name="text">
                <string>0%...100%</string>
               </property>
              </widget>
             </item>
             <item row="0" column="0">
              <widget class="QPushButton" name="pbSlotPatternFull">
               <property name="text">
                <string>All 100%</string>
               </property>
              </widget>
             </item>
             <item row="1" column="2">
              <widget class="QPushButton" name="pbSlotPatternDecreasing">
               <property name="text">
                <string>100%...0%</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QPushButton" name="pbSlotPatternOdd">
               <property name="text">
                <string>Odd 100%</string>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QPushButton" name="pbSlotPatternEven">
               <property name="text">
                <string>Even 100%</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item row="0" column="0">
           <layout class="QFormLayout" name="layoutSlotSetup">
            <item row="0" column="0">
             <widget class="QLabel" name="startCodeLabel">
              <property name="text">
               <string>Start Code</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QSpinBox" name="startCodeEdit">
              <property name="maximum">
               <number>255</number>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="dataSlotCountLabel">
              <property name="text">
               <string>Data Slot Count</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QSpinBox" name="dataSlotCountEdit">
              <property name="maximum">
               <number>512</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="1" column="0" colspan="4">
           <widget class="QTableView" name="dataSlotsTable">
            <property name="selectionMode">
             <enum>QAbstractItemView::SelectionMode::SingleSelection</enum>
            </property>
           </widget>
          </item>
          <item row="0" column="3">
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Orientation::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>801</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>&amp;Help</string>
    </property>
    <addaction name="actionAboutQt"/>
   </widget>
   <addaction name="menuHelp"/>
  </widget>
  <action name="actionAboutQt">
   <property name="text">
    <string>About &amp;Qt</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
